docs/rules/require-default-props.md

```diff
@@ -197,13 +197,18 @@ NotAComponent.propTypes = {
 
 ```js
 ...
-"react/require-default-props": [<enabled>, { forbidDefaultForRequired: <boolean>, ignoreFunctionalComponents: <boolean> }]
+"react/require-default-props": [<enabled>, {
+  "forbidDefaultForRequired": <boolean>,
+  "ignoreFunctionalComponents": <boolean>,
+  "forceDefaultPropArguments": <boolean>
+}]
 ...
 ```
 
-* `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
-* `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to false.
-* `ignoreFunctionalComponents`: optional boolean to ignore this rule for functional components. Defaults to false.
+- `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
+- `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to false.
+- `ignoreFunctionalComponents`: optional boolean to ignore this rule for functional components. Defaults to false.
+- `forceDefaultPropArguments`: optional booolean to assign default to object instead of `defaultProps` for functional components. Defaults to false.
```

Comment:
i think we should make sure that `forceDefaultPropArguments` can not be set to `true` when `ignoreFunctionalComponents` is `true`?

or maybe, we should deprecate `ignoreFunctionalComponents`, and come up with an alternate schema:
```
{
  "forbidDefaultForRequired": <boolean>,

  // for now, this wouldn't support any other choices, and would be optional, but later could be set to "ignore" as long as `functions` wasn't set to "ignore"
  "classes": "defaultProps",

  "functions": "ignore" | "defaultProps" | "default arguments"
}
```

Thoughts?

Reply:
> we should deprecate ignoreFunctionalComponents, and come up with an alternate schema

I like this, then should the schema be like following lines? 

```
{
  "forbidDefaultForRequired": <boolean>,
  "classes": "defaultProps",
  "functions": "ignore" | "defaultProps" | "defaultArguments"
}
```

Or should I keep the `forbidDefaultForRequired`?

Reply:
Yep, exactly. And the new stuff would only be permitted if ignoreFunctionalComponents is omitted.